index.js

```diff
@@ -168,7 +180,34 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
   async submitPassword(password) {
+    // If a user is trying to submit a password, we should lock
+    // in the event that the password is incorrect
+    this.setLocked();
```

Comment:
What motivated this change? We should only be calling `submitPassword` when the keyring is already locked, so this might not be entirely necessary.

I ask because this is introducing a new state update, as a side-effect of `setLocked`, and I'm not sure what effect it will have. And to make matters worse, there is an asynchronous step in `setLocked` which could cause confusing (the `lock` event might end up being emitted after unlocked, which is confusing).

Reply:
It's not something I consider absolutely necessary but allowing to submit password, while already unlocked and populated, seems weird.  Like, if you go to a login page but you're already logged in, you get redirected to the home screen.  Like I said, not strictly necessary, but seems something should happen here.

Reply:
I've removed this, as it isn't strictly related to this PR.

Reply:
Another approach to consider would be adding a `try...catch`, and locking if we get an error. That would avoid any misleading `lock` event in the happy path